overlord/servicestate/quota_handlers.go

```diff
@@ -46,27 +45,27 @@ import (
 // modification that lives in a task.
 type QuotaControlAction struct {
 	// QuotaName is the name of the quota group being controlled.
-	QuotaName string `json:"quota-name"`
+	QuotaName string `json:"quota-name,omitempty"`
 
 	// Action is the action being taken on the quota group. It can be either
 	// "create", "update", or "remove".
-	Action string `json:"action"`
+	Action string `json:"action,omitempty"`
 
 	// AddSnaps is the set of snaps to add to the quota group, valid for either
 	// the "update" or the "create" actions.
-	AddSnaps []string `json:"snaps"`
+	AddSnaps []string `json:"snaps,omitempty"`
 
-	// MemoryLimit is the memory limit for the quota group being controlled,
-	// either the initial limit the group is created with for the "create"
+	// ResourceLimits is the set of resource limits to set on the quota group.
+	// Either the initial limit the group is created with for the "create"
 	// action, or if non-zero for the "update" the memory limit, then the new
 	// value to be set.
-	MemoryLimit quantity.Size
```

Comment:
This feels like there is a small complication here that we probably need to handle. AIUI this QuotaControlAction  is part of the `quota-control-actions` task. So I think what **could** happen is that a change with "quota-control-actions" is generated by the old snapd that has MemoryLimit in there. Now in parallel a snapd refresh is happening and the new snapd starts and processes the QuotaControlAction struct and now expects a ResourceLimits struct here.

I think there are two ways to solve this:
a) keep `MemoryLimit` as a compatibility in the struct and deal with it
b) create a `patch`, see `overlord/patch/` that converts any "in-flight" changes of quota-control-action from MemoryLimit to ResourceLimits. iirc patch6_2.go does similar things.

I **could** be wrong about this, it's been a while and maybe the quota changes/tasks are special somehow so that this is not a concern. Also if true you could consider splitting this particular change into it's own PR for quicker/easier review.

Also any json name change in this struct will have to be converted in the same way.

Alternatively we could keep the names in this PR and instead of adding "ResourceLimits" here we keep "MemoryLimit" and do these changes in a followup.

Reply:
@mvo5 As I mentioned in my earlier review I personally think this is okay to ignore this problem since the feature is still experimental and the likelihood of a snapd refresh happening at the same time as a quota-control-action. Of course if you really think we need to handle this, then we will need to do as you mentioned but IMHO it's not worth it

Reply:
Sorry, I missed that this was already discussed. Given that it's experimental I agree.